### DISCUSSION TITLE
[macOS] Associate app with supported filetypes

### DISCUSSION THREAD

**Discussion by gingerbeardman** [Orama-Interactive/Pixelorama#582] [Ideas]
I'd like to be able to right click an image in Finder (macOS file manager) and choose Open in... and then choose Pixelorama.

This code should be helpful to implement the opening of a passed file:

https://github.com/godotengine/godot/pull/15625/files

The app bundle plist will also need to be amended to include supported file types (as Uniform Type Identifier) so the OS knows which types Pixelorama is OK to open.

**Comment by OverloadedOrama** [comment]
This is already possible in Windows, thanks to `OS.get_cmdline_args()`. You can assossiate .pxo files and image files and have them open in Pixelorama. You can also drag and drop them onto the executable file, and in v0.8 you will be able to drag and drop files inside Pixelorama, while it's running, to open them. Is it different in macOS? Do we somehow have to edit the `Info.plist` file, or is there an extra setting in Godot to fix this?

**Comment by gingerbeardman** [comment]
Ah, my use case is to load PNGs and other non-PXO files.

I'm not familiar with Godot but I'll try to look into the plist thing. How is the Mac app bundle generated?

**Comment by OverloadedOrama** [comment]
I'm not sure how Godot handles the app bundle generation behind the scenes. All I just do is just export the project to macOS. In Windows, "open with" works with non-pxo files too.

**Comment by gingerbeardman** [comment]
I'll see if I can find more info. Thanks!
